### PR TITLE
Add ADR for the HTTPClient rework

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

File: docs/decisions/0002-manage-http-client-in-core.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Manage the HTTPClient in the core
+
+## Why
+
+At the moment, the core declare a Java interface for HttpClient, that should be implemented by each client, and passed to several APIs. It is the responsibility of the client side to implement this interface, by choosing an HTTPClient implementation, and configuring it (SSL, proxy, timeouts, credentials, ...).
```
Comment (nquinquenel) [inline docs/decisions/0002-manage-http-client-in-core.md:5]:
```suggestion
At the moment, the core declares a Java interface for HttpClient, that should be implemented by each client, and passed to several APIs. It is the responsibility of the client side to implement this interface, by choosing an HTTPClient implementation, and configuring it (SSL, proxy, timeouts, credentials, ...).
```

**Review by nquinquenel — APPROVED** [review]
LGTM

File: docs/decisions/0002-manage-http-client-in-core.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+* For SLVSCODE, there should be almost no impact
+* For SLE, it will change the HTTPClient from OkHttp to Apache HTTPClient, so it could have some side effects
+* For SLI, this will be the same library, but the integration with IntelliJ SSL and Proxy settings will need to be reworked with care
+* The HTTPClient interface will be deprecated, and should ultimately disappear
+
+## How
+
+New requests will be added to the backend protocol:
+
+* `getCrendetials` to ask client for credentials of a given connection
```
Comment (damien-urruty-sonarsource) [inline docs/decisions/0002-manage-http-client-in-core.md:55]: getCrendetials -> getCredentials

**Review by damien-urruty-sonarsource — APPROVED** [review]
Good job!

**Pull request merged**